cumulusci/core/keychain/base_project_keychain.py

```diff
@@ -201,64 +154,138 @@ def get_org(self, name: str):
-            self._raise_service_not_valid(name)
-        if name not in self.services:
-            if name == "connected_app":
+        if not self.project_config.services:
+            raise ServiceNotValid(
+                "Expecting services to be loaded, but none were found."
+            )
+        elif service_type not in self.project_config.services:
+            raise ServiceNotConfigured(
+                f"Service type is not configured: {service_type}"
+            )
+
+        if service_type not in self.services:
+            if service_type == "connected_app":
                 return DEFAULT_CONNECTED_APP
-            self._raise_service_not_configured(name)
+            self._raise_service_not_configured(service_type)
+
+        if not alias:
+            alias = self._default_services.get(service_type)
+            if not alias:
+                raise CumulusCIException(
+                    f"No default service currently set for service type: {service_type}"
+                )
+        service = self._get_service(service_type, alias)
 
-        service = self._get_service(name)
         # transparent migration of github API tokens to new key
-        if name == "github" and service.password and not service.token:
+        if service_type == "github" and service.password and not service.token:
             service.config["token"] = service.password
+
         return service
 
-    def _get_service(self, name):
-        return self.services.get(name)
+    def _get_service(self, service_type, alias):
+        return self.services[service_type][alias]
 
-    def _validate_key(self):
-        pass
+    def _validate_service(self, service_type, alias, config):
+        self._validate_service_alias(service_type, alias)
+        self._validate_service_attributes(service_type, config)
 
-    def _validate_service(self, name, service_config):
+    def _validate_service_attributes(self, service_type, service_config):
+        """
+        Validate that all of the required attributes for a
+        given service are present.
+        """
         missing_required = []
-        attr_key = f"services__{name}__attributes"
+        attr_key = f"services__{service_type}__attributes"
         for atr, config in list(getattr(self.project_config, attr_key).items()):
-            if config.get("required") is True and not getattr(service_config, atr):
+            if config.get("required") and not getattr(service_config, atr):
                 missing_required.append(atr)
 
         if missing_required:
-            self._raise_service_not_valid(name)
+            raise ServiceNotValid(
+                f"Missing required attribute(s) for service: {missing_required}"
+            )
+
+    def _validate_service_alias(self, service_type, alias):
+        if alias == service_type:
+            raise ServiceNotValid(
+                "Service alias cannot be the same as the service type."
+            )
```

Comment:
Is there a reason we need this restriction or is it just to avoid confusion?

Reply:
This was simply to avoid confusion.